app/scripts/lib/rpc-method-middleware/handlers/watch-asset.js

```diff
@@ -39,6 +41,21 @@ async function watchAssetHandler(
       params: { options: asset, type },
       origin,
     } = req;
+
+    const { tokenId } = asset;
+
+    if (
+      [ERC721, ERC1155].includes(type) &&
+      tokenId &&
+      typeof tokenId !== 'string'
```

Comment:
this would allow
```
{
  "type": "ERC721",
  "asset": {
    "tokenId": 0
  }
}
```

Reply:
Good catch. Fixed [here](https://github.com/MetaMask/metamask-extension/pull/19738/commits/b5267995d186c4db2e7b2f9c84b208e7f9209493)